platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java

```diff
@@ -135,6 +134,11 @@ public void activate() {
   }
 
   public void deactivate() {
+    if (activeCounter == 0) {
```

Comment:
It seems that we should deal with these multiple deactivations at the source (MapSnapshotter). Otherwise, if `activate` / `deactivate` is not called in pairs, we have no solid state tracking for the activation state and it could be deactivated before all dependents are actually ready for deactivation.